src/Components/Partner-Deals/Modals/Agreement-Upload/agreementUpload.component.js

```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react';
+import './agreementUpload.component.css';
+
+import { Get, Post } from 'common-js-util';
+import { API_HOST } from './../../../../Constants/global.constants';
+import SelectBox from './../../../Forms/Components/Select-Box/selectBoxForGenericForm.component';
+import { ModalManager } from 'drivezy-web-utils/build/Utils/modal.utils';
+import ImageUpload from './../../../Forms/Components/Image-Upload/imageUpload.component';
+
+export default class AgreementUpload extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            agreement: "",
+            business: "",
+            agreement_type: [],
+            business_status: [],
+            parent: this.props.parent,
+            files: []
+        }
+    }
+
+    componentWillMount() {
+        this.getAgreementType();
+    }
+
+    getAgreementType = async () => {
+        const result = await Get({
+            url: "lookupValue?query=lookup_type=117&limit=100",
+            urlPrefix: API_HOST
+        });
+        if (result.success) {
+            this.setState({ agreement_type: result.response });
+        }
+        this.getBusinessStatus();
+    }
+
+    getBusinessStatus = async () => {
+        const result = await Get({
+            url: "lookupValue?query=lookup_type=118&limit=100",
+            urlPrefix: API_HOST
+        });
+        if (result.success) {
+            this.setState({ business_status: result.response });
+        }
+    }
+
+    vehiclePayment = async() => {
+        const { parent, transaction_amount, payment_mode, transaction_id } = this.state;
+        // console.log(parent)
+        const result = await Post({
+            url: "vehiclePayment",
+            body: {
+                amount: transaction_amount,
+                deal_id: parent.id,
+                payment_mode_id: payment_mode,
+                payment_mode_type: parent.financing_type_id, //not sure about mode type id
+                transaction_id: transaction_id
+            },
+            urlPrefix: API_HOST
+        });
+        if (result.success) {
+            console.log(result);
```

Comment:
What is this for?

Reply:
Have to see for the uploaded files then will change